tests/apps/riskofbias/test_riskofbias_api.py

```diff
@@ -40,6 +40,21 @@ def test_riskofbias_detail(db_keys):
     ]
 
 
+@pytest.mark.django_db
```

Comment:
we need a test on the renderer, not a specific API endpoint: Use this instead:

```python
from rest_framework import status

def test_base_renderer_not_success():
    data = {"error": "Method not allowed"}
    context = {"response": Response(status=status.HTTP_405_METHOD_NOT_ALLOWED)}
    result = renderers.PandasXlsxRenderer().render(data, renderer_context=context)
    assert result == json.dumps(data)

```

Reply:
we should also throw an error in the base renderer if status is not ok and the response isn't json, since we don't know how to return

Reply:
I originally wrote something very similar to that, but it was not reproducing the server error. It looks like when the view calls ```response.render(); response.rendered_content``` the error is triggered. The view also sets a bunch of properties on that response before the call to render, which I tried to write myself but the code was losing some readability.

Reply:
I wrote a test for the xlsx renderer that replicates the server error.